app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java

```diff
@@ -34,11 +34,28 @@ protected static String buildUrl(String[] ingridients) {
     }
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_choose_ingredients);
-        Toolbar toolbar = findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
+        setContentView(R.layout.content_choose_ingredients);
+
+        final GridView gridview = (GridView) findViewById(R.id.gridview);
```

Comment:
can be extracted to outer method "setupGridView",
will make `onCreate` cleaner.